jdk8 without javac

Hi I've been ignoring it for a while but now 
I would need javac and it's missing from the 
package recently. It used to be included 
some months ago and it worked perfectly.

That's a good hint yes, I even tried it 
but it's probably something else, not 
even libs, I just can't elaborate, 
here's the point of failure:
```
gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I..  -DLOCALEDIR=\"/usr/local/share/locale\" -DSYSCONFDIR=\"/usr/local/etc\" -I../lib -I../lib -I/usr/local/include/ncurses  -g -O2 -Wall -MT cut.o -MD -MP -MF .deps/cut.Tpo -c -o cut.o cut.c
color.c: In function ‘color_update’:
color.c:230:2: error: unknown type name ‘magic_t’
  magic_t cookie = NULL;
[ etc... ]
make[1]: Leaving directory '/usr/local/tmp/crew/nano-2.8.4.tar.xz.dir/nano-2.8.4'
make: *** [Makefile:1470: all] Error 2
nano failed to install: `make -j2` exited with 2
```

I couldn't be more off-topic though, 
so i'm only wondering if anyone else 
has this issue with jdk8. 
I'd paste my crew installation log as an only clue, if I may: 
```
chronos@localhost / $ crew install jdk8
jdk8: The JDK is a development environment for building applications, applets, and components using the Java programming language.
http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html
version 8u131
No precompiled binary available for your platform, downloading source...
--2017-07-29 17:43:55--  http://javadl.oracle.com/webapps/download/AutoDL?BundleId=220305_d54c1d3a095b4ff2b6607d096fa80163
Resolving javadl.oracle.com... 137.254.120.23
Connecting to javadl.oracle.com|137.254.120.23|:80... connected.
HTTP request sent, awaiting response... 302 Moved Temporarily
Location: http://sdlc-esd.oracle.com/ESD6/JSCDL/jdk/8u131-b11/d54c1d3a095b4ff2b6607d096fa80163/jre-8u131-linux-x64.tar.gz?GroupName=JSC&FilePath=/ESD6/JSCDL/jdk/8u131-b11/d54c1d3a095b4ff2b6607d096fa80163/jre-8u131-linux-x64.tar.gz&BHost=javadl.sun.com&File=jre-8u131-linux-x64.tar.gz&AuthParam=1501344238_4e64b9b58fe652bc0534abc0a9fb8eea&ext=.gz [following]
--2017-07-29 17:43:56--  http://sdlc-esd.oracle.com/ESD6/JSCDL/jdk/8u131-b11/d54c1d3a095b4ff2b6607d096fa80163/jre-8u131-linux-x64.tar.gz?GroupName=JSC&FilePath=/ESD6/JSCDL/jdk/8u131-b11/d54c1d3a095b4ff2b6607d096fa80163/jre-8u131-linux-x64.tar.gz&BHost=javadl.sun.com&File=jre-8u131-linux-x64.tar.gz&AuthParam=1501344238_4e64b9b58fe652bc0534abc0a9fb8eea&ext=.gz
Resolving sdlc-esd.oracle.com... 104.83.4.11, 104.83.4.26
Connecting to sdlc-esd.oracle.com|104.83.4.11|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 75920010 (72M) [application/x-gzip]
Saving to: 'AutoDL'

AutoDL                              100%[================================================================>]  72.40M  2.40MB/s    in 30s     

2017-07-29 17:44:26 (2.43 MB/s) - 'AutoDL' saved [75920010/75920010]

Archive downloaded
Unpacking archive, this may take a while...
gcc (GCC) 4.9.4
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Building from source, this may take a while...
Preconfiguring package...
Installing...
Jdk8 installed!
chronos@localhost / $ which java
/usr/local/bin/java
chronos@localhost / $ which javac
which: no javac in (/usr/local/bin:/usr/bin:/bin:/opt/bin)
```

--- a/packages/jdk8.rb
+++ b/packages/jdk8.rb
@@ -3,25 +3,40 @@ require 'package'
 class Jdk8 < Package
   description 'The JDK is a development environment for building applications, applets, and components using the Java programming language.'
   homepage 'http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html'
-  version '8u131'
-  case ARCH
-  when 'i686'
-    source_url 'http://javadl.oracle.com/webapps/download/AutoDL?BundleId=220303_d54c1d3a095b4ff2b6607d096fa80163'
-    source_sha256 'a773f2fe17061ef637ed2094b06313a99c0b45ba3d3cb7f8f1ebf18448495aeb'
-  when 'x86_64'
-    source_url 'http://javadl.oracle.com/webapps/download/AutoDL?BundleId=220305_d54c1d3a095b4ff2b6607d096fa80163'
-    source_sha256 '355e5cdb066d4cada1f9f16f358b6fa6280ff5caf7470cf0d5cdd43083408d35'
-  when 'armv7l'
-    source_url 'https://www.dropbox.com/s/vcejuitboafaxib/jdk8u22-armv7l.tar.gz'
-    source_sha256 'be13670ce0588a888190a55a63a4a95940b8cd77f6dea3dfeaefe3a9ed800c0b'
-  when 'aarch64'
-    source_url 'https://www.dropbox.com/s/vcejuitboafaxib/jdk8u22-armv7l.tar.gz'
-    source_sha256 'be13670ce0588a888190a55a63a4a95940b8cd77f6dea3dfeaefe3a9ed800c0b'
-  end
+  version '8u144'
+  source_url 'http://hg.openjdk.java.net/jdk8/jdk8/archive/2a8f4c022aa0.tar.gz'
+  source_sha256 'e632ae353ccfb90957cbb6a5818ee063710a41e7b97b490ee58ca5f627d863f1'
+
   def self.install
+    case ARCH
+    when 'aarch64'
+      system 'wget -c --header "Cookie: oraclelicense=accept-securebackup-cookie" --no-check-certificate https://download.oracle.com/otn-pub/java/jdk/8u144-b01/090f390dda5b47b9b721c7dfaa008135/jdk-8u144-linux-arm32-vfp-hflt.tar.gz'
+      abort 'Checksum mismatch. :/ Try again.'.lightred unless Digest::SHA256.hexdigest( File.read('jdk-8u144-linux-arm32-vfp-hflt.tar.gz') ) == 'cbbd390e19ab4c473e05f60602ce2804db25e4e35be5ab95f4f1a2aeb5b72383'
+      system 'tar xvf jdk-8u144-linux-arm32-vfp-hflt.tar.gz'
+    when 'armv7l'
+      system 'wget -c --header "Cookie: oraclelicense=accept-securebackup-cookie" --no-check-certificate https://download.oracle.com/otn-pub/java/jdk/8u144-b01/090f390dda5b47b9b721c7dfaa008135/jdk-8u144-linux-arm32-vfp-hflt.tar.gz'
+      abort 'Checksum mismatch. :/ Try again.'.lightred unless Digest::SHA256.hexdigest( File.read('jdk-8u144-linux-arm32-vfp-hflt.tar.gz') ) == 'cbbd390e19ab4c473e05f60602ce2804db25e4e35be5ab95f4f1a2aeb5b72383'
+      system 'tar xvf jdk-8u144-linux-arm32-vfp-hflt.tar.gz'
+    when 'i686'
+      system 'wget -c --header "Cookie: oraclelicense=accept-securebackup-cookie" --no-check-certificate https://download.oracle.com/otn-pub/java/jdk/8u144-b01/090f390dda5b47b9b721c7dfaa008135/jdk-8u144-linux-i586.tar.gz'
+      abort 'Checksum mismatch. :/ Try again.'.lightred unless Digest::SHA256.hexdigest( File.read('jdk-8u144-linux-i586.tar.gz') ) == '624c090647629394ef0ee08d9d8ac5d3d5a9a60fa245fefb2eb417c36c7cb7c4'
+      system 'tar xvf jdk-8u144-linux-i586.tar.gz'
+    when 'x86_64'
+      system 'wget -c --header "Cookie: oraclelicense=accept-securebackup-cookie" --no-check-certificate https://download.oracle.com/otn-pub/java/jdk/8u144-b01/090f390dda5b47b9b721c7dfaa008135/jdk-8u144-linux-x64.tar.gz'
+      abort 'Checksum mismatch. :/ Try again.'.lightred unless Digest::SHA256.hexdigest( File.read('jdk-8u144-linux-x64.tar.gz') ) == 'e8a341ce566f32c3d06f6d0f0eeea9a0f434f538d22af949ae58bc86f2eeaae4'
+      system 'tar xvf jdk-8u144-linux-x64.tar.gz'
+    end
     system "mkdir -p #{CREW_DEST_DIR}/usr/local"
-    system "cp -r bin/ #{CREW_DEST_DIR}/usr/local"
-    system "cp -r lib/ #{CREW_DEST_DIR}/usr/local"
-    system "cp -r man/ #{CREW_DEST_DIR}/usr/local"
+    FileUtils.cd('jdk1.8.0_144') do
+      system "cp -r bin/ #{CREW_DEST_DIR}/usr/local"
+      system "cp -r lib/ #{CREW_DEST_DIR}/usr/local"
+      system "cp -r db/bin/ #{CREW_DEST_DIR}/usr/local"
+      system "cp -r db/lib/ #{CREW_DEST_DIR}/usr/local"
+      system "cp -r include/ #{CREW_DEST_DIR}/usr/local"
+      system "cp -r jre/bin/ #{CREW_DEST_DIR}/usr/local"
+      system "cp -r jre/lib/ #{CREW_DEST_DIR}/usr/local"
+      system "cp -r jre/plugin/ #{CREW_DEST_DIR}/usr/local"
+      system "cp -r man/ #{CREW_DEST_DIR}/usr/local"
+    end
   end
 end

--- a/packages/jdk8.rb
+++ b/packages/jdk8.rb
@@ -26,17 +26,14 @@ class Jdk8 < Package
       abort 'Checksum mismatch. :/ Try again.'.lightred unless Digest::SHA256.hexdigest( File.read('jdk-8u144-linux-x64.tar.gz') ) == 'e8a341ce566f32c3d06f6d0f0eeea9a0f434f538d22af949ae58bc86f2eeaae4'
       system 'tar xvf jdk-8u144-linux-x64.tar.gz'
     end
-    system "mkdir -p #{CREW_DEST_DIR}/usr/local"
+    system "mkdir -p #{CREW_DEST_DIR}#{CREW_LIB_PREFIX}/jdk8"
     FileUtils.cd('jdk1.8.0_144') do
-      system "cp -r bin/ #{CREW_DEST_DIR}/usr/local"
-      system "cp -r lib/ #{CREW_DEST_DIR}/usr/local"
-      system "cp -r db/bin/ #{CREW_DEST_DIR}/usr/local"
-      system "cp -r db/lib/ #{CREW_DEST_DIR}/usr/local"
-      system "cp -r include/ #{CREW_DEST_DIR}/usr/local"
-      system "cp -r jre/bin/ #{CREW_DEST_DIR}/usr/local"
-      system "cp -r jre/lib/ #{CREW_DEST_DIR}/usr/local"
-      system "cp -r jre/plugin/ #{CREW_DEST_DIR}/usr/local"
-      system "cp -r man/ #{CREW_DEST_DIR}/usr/local"
+      system "rm -f src.zip"
+      system "cp -r * #{CREW_DEST_DIR}#{CREW_LIB_PREFIX}/jdk8"
+    end
+    system "mkdir -p #{CREW_DEST_DIR}#{CREW_PREFIX}/bin"
+    FileUtils.cd("#{CREW_DEST_DIR}#{CREW_PREFIX}/bin") do
+      system "ln -s ../../../#{CREW_LIB_PREFIX}/jdk8/bin/* ."
     end
   end
 end

